reason-parser/src/reason_parser.mly

```diff
@@ -1200,7 +1201,7 @@ _use_file:
 parse_core_type:
     core_type EOF
       {
-        let core_loc = mklocation $startpos($1) $endpos($1) in
+        let core_loc = mklocation $symbolstartpos $endpos($1) in
```

Comment:
`$symbolstartpos` is determined locally. Exactly the same with attributes: an empty sequence is an optional (nullable producer) that isn't there.

To determine symbolstartpos:
```
let rec symbolstartpos i =
  if i >= producers then $endpos
  else if null($i) then symbolstartpos (i+1)
  else $startpos($i)
in
symbolstartpos 0
```

The code generated is a bit more efficient, by unrolling the loop manually up to the first non-nullable producer.
But the main point is that this is purely local: it doesn't do anything that you couldn't by manually manipulating `$startpos($i)`